dot/parachain/backing/candidate_backing.go

```diff
@@ -49,6 +50,17 @@ type CandidateBacking struct {
 	// This is guaranteed to have an entry for each candidate with a relay parent in the implicit
 	// or explicit view for which a `Seconded` statement has been successfully imported.
 	perCandidate map[parachaintypes.CandidateHash]*perCandidateState
+	// State tracked for all active leaves, whether or not they have prospective parachains enabled.
+	perLeaf map[common.Hash]ActiveLeafState
+	// The utility for managing the implicit and explicit views in a consistent way.
+	// We only feed leaves which have prospective parachains enabled to this view.
+	implicitView ImplicitView
+}
+
+type ActiveLeafState struct {
+	ProspectiveParachainsMode parachaintypes.ProspectiveParachainsMode
+	SecondedAtDepth           map[parachaintypes.ParaID]btree.Map[uint, parachaintypes.CandidateHash]
+	perCandidate              map[parachaintypes.CandidateHash]*perCandidateState //nolint:unused
```

Comment:
Why is the value a pointer?

Reply:
To quickly modify the value.
Otherwise, we will have to change the local variable and then add it to the map.

Example without pointer value:
```
type group struct {
	people map[int]person
}
func (g group) changePersonName(id int, name string) {
	p := g.people[id]
	p.name = name

	g.people[id] = p // add modified local variable to map
}
```

Example with pointer value:
```
type group struct {
	people map[int]*person
}
func (g group) changePersonName(id int, name string) {
	p := g.people[id]
	p.name = name
}
```

Reply:
In that case, we need to have nil checks on the value. 

There is one instance in `candidate_backing.go` line 323 where it has no nil checks. Please make sure it is handled everywhere.

Reply:
Okay. 
It's not from this PR, but I will add a nil check there.